forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -538,8 +538,10 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
                     if (repres != ReplacementResult.NotReplaced) continue;
                 }
                 if (card == c) {
+                    storeChangesZoneAll(copied, zoneFrom, zoneTo, params);
                     zoneTo.add(copied, position, toBattlefield ? null : lastKnownInfo); // the modified state of the card is also reported here (e.g. for Morbid + Awaken)
                 } else {
+                    storeChangesZoneAll(card, zoneFrom, zoneTo, params);
```

Comment:
The mutate handling isn't perfect - you added the components here: a653085

most likely so it works with returning all parts via UntilHostLeavesPlay?

but I think _Hero of Bretagard_ should probably only gain 1 counter when a mutated permanent gets exiled

So it might better be done inside TriggerHandler like for ChangesZone here:
https://github.com/Card-Forge/forge/blob/acb5151a8289d788afaa9653e79cf507a85ddc0e/forge-game/src/main/java/forge/game/trigger/TriggerHandler.java#L467

but maybe something for later since it would require even more refactor

Reply:
"or a spell or ability you control exiles one or more permanents from the battlefield" imo that need to use LKI ?

meaning it should only see one Mutated Card?

Hero of Bretagard probably need to be refactored into an Exile Trigger anyway instead of a ChangesZones Trigger

Reply:
Yea the wording is sus, we'd need new ExiledAll but for some reason it doesn't have a ruling to ignore RE like _Soulherder_

Reply:
yeah, but i think it is still implied

also for touching Exile, i need to add a Static/Replacment for Exile GameAction to add:
"Cause can't cause you to exile stuff" for the WHO card

that is on my TODO list

Reply:
Hero of Bretagard  should be updated in a later MR too